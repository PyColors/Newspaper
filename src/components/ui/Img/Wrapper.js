import styled from 'styled-components';

const Wrapper = styled.figure`
  min-height: 250px;
`;

export default Wrapper;
