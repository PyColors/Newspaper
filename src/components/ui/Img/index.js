import React from 'react';
import PropTypes from 'prop-types';

import Wrapper from './Wrapper';

const Img = props => {
  return (
    <Wrapper>
      <img
        src={props.src}
        alt={props.alt}
        title={props.title}
        width={props.width}
        height={props.height}
      />
    </Wrapper>
  );
};

Img.propTypes = {
  alt: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
  title: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number
};

export default Img;
