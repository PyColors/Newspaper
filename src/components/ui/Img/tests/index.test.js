import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Img from '../index';

const src = 'src.png';
const alt = 'description';

const renderComponent = (props = {}) =>
  shallow(<img src={src} alt={alt} {...props} />);

describe('<Img />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(<Img src="http://pycolor.com/test.jpg" alt="PyColors" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have an src attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('src')).toEqual(src);
  });

  it('should have an alt attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('alt')).toEqual(alt);
  });

  it('should render its text', () => {
    const children = 'Text';
    const renderedComponent = shallow(<Img>{children}</Img>);
    expect(renderedComponent.contains(children)).toBe(false);
  });

  it('should not have a className attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('className')).toBeUndefined();
  });

  it('should assume a className attribute', () => {
    const className = 'test';
    const renderedComponent = renderComponent({ className });
    expect(renderedComponent.hasClass(className)).toBe(true);
  });
});
