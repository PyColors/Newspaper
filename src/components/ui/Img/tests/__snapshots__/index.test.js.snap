// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> should match the snapshot 1`] = `
<figure
  className="sc-bdVaJa EZFKr"
>
  <img
    alt="PyColors"
    className="sc-bwzfXH fvJsht"
    height={undefined}
    src="http://pycolor.com/test.jpg"
    title={undefined}
    width={undefined}
  />
</figure>
`;
